showcase/tests/integration/components/hds/flyout/index-test.js

```diff
@@ -148,7 +150,7 @@ module('Integration | Component | hds/flyout/index', function (hooks) {
     await render(
       hbs`<Hds::Flyout id="test-flyout" as |F|><F.Header>Title</F.Header></Hds::Flyout>`
     );
-    assert.dom('button.hds-flyout__dismiss').isFocused();
+    assert.dom('button.hds-dialog-primitive__dismiss').isFocused();
```

Comment:
This was failing because the `@extraClass` was not passed down. Once that is added, you can undo these changes in this test.

Reply:
_**TODO:** (for Kristin) revert change to test_

Reply:
I'll have to refactor the `@extraClass` to use as a variable for the "component" name (such as "dialog-primitive", "flyout", or "modal")...

Should probably rename it then too since then the name "extraClass" won't be clear. Maybe "componentName"?

Reply:
Actually maybe it can be simpler. Still need to work through it...

Reply:
I think maybe we could move the @extraClass to the container index.hds and then pass it down to the child components so for example if you set the extraClass to "hds-flyout" it could get appended to child component classes.

Or am I thinking about this wrong? @didoo

Reply:
Yes, `extraClass` is a way to bypass a limitation in how Ember does the `yield` in templates, and pass an extra class name to a (yielded) component. It's not a way to provide classnames prefixes or similar.

No looking at the suggestion above, I may have added the comment in the wrong place: this is a button, so there's no way to pass a custom extraClass to it. Sorry for the conclusion.